packages/plugin-workspace-tools/sources/commands/focus.ts

```diff
@@ -35,7 +40,9 @@ export default class WorkspacesFocus extends BaseCommand {
     const cache = await Cache.find(configuration);
 
     let requiredWorkspaces: Set<Workspace>;
-    if (this.workspaces.length === 0) {
+    if (this.all) {
```

Comment:
Might be a good idea to also add some validation via a schema that makes `this.all` and `this.workspaces.length > 0` exclusive (since it wouldn't make sense to specify workspaces when using the `-A,--all` flag).